sympy/solvers/solveset.py

```diff
@@ -192,6 +191,11 @@ def inv(trig):
                 invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
             return _invert_real(f.args[0], invs, symbol)
 
+    if (isinstance(a,log) for a in f.args):
```

Comment:
I felt the same problem. Should I remove the if condition @jksuom ?
It can work without it,I just wanted to check if there is any `log` present or not ?